tools/windows-node-installer/pkg/cloudprovider/factory.go

```diff
@@ -25,6 +54,25 @@ func CloudProviderFactory(kubeconfigPath, credentialPath, credentialAccountID, r
 	}
 
 	switch provider := cloudProvider.Type; provider {
+	case "Azure":
+		//Gather the azure related information.
+		//Read the .auth file to get the subscrptionID and other details.
+		data, err := readJSON("azureauth_location")
+		template, err := readJSON("windowsnode-template.json")
+		if err != nil {
+			log.Fatal("Unable to read template file")
```

Comment:
Please remove `[log.Fatal()](https://golang.org/pkg/log/#Fatal)` calls in all cases. They will cause the program to exit making your next return statement meaningless. It is also a bad pattern. Only the main function should end up calling `log.Fatal()`

Reply:
will fix that.